apm-agent-core/src/test/java/co/elastic/apm/agent/report/serialize/DslJsonSerializerTest.java

```diff
@@ -425,6 +425,7 @@ void testSpanDestinationContextSerialization() {
         span.getContext().getDestination().withAddress("whatever.com").withPort(80)
             .getService()
             .withResource("whatever.com:80");
+        span.getContext().getDestination().getCloud().withRegion("us-east-1");
```

Comment:
Because the JSON writing output is multi-branched (`hasPort || hasServiceContent || hasCloudContent`) with each of combination of true/false on these taking a different path especially on how objects terminate, I think you need to expand the testing to multiple combinations of these. Otherwise it would be easy for a maintenance change to result in the wrong JSON termination in edge cases, better to catch that here